src/functions.rs

```diff
@@ -961,6 +961,75 @@ fn call_to_fsc_rules<'a>(
     Ok(ret)
 }
 
+fn call_to_sids<'a>(
+    c: &FuncCall,
+    types: &TypeMap,
+    class_perms: &ClassList,
+    context: &BlockContext<'_>,
+    file: &SimpleFile<String, String>,
+) -> Result<Vec<Sid<'a>>, CascadeErrors> {
+    if context.in_function_block() {
+        return Err(CascadeErrors::from(
+            ErrorItem::make_compile_or_internal_error(
+                "initial_context() calls are not valid in functions",
+                Some(file),
+                c.name.get_range(),
+                "Remove this call",
+            ),
+        ));
+    }
+
+    let target_args = vec![
+        FunctionArgument::new(
+            &DeclaredArgument {
+                param_type: CascadeString::from("string"),
+                is_list_param: false,
+                name: CascadeString::from("sid_name"),
+                default: None,
+            },
+            types,
+            None,
+        )?,
+        FunctionArgument::new(
+            &DeclaredArgument {
+                param_type: CascadeString::from("context"),
+                is_list_param: false,
+                name: CascadeString::from("fs_label"),
+                default: None,
+            },
+            types,
+            None,
+        )?,
+    ];
+    let validated_args =
+        validate_arguments(c, &target_args, types, class_perms, context, Some(file))?;
+    let mut args_iter = validated_args.iter();
+
+    let sid_name = args_iter
+        .next()
+        .ok_or_else(|| ErrorItem::Internal(InternalError::new()))?
+        .get_name_or_string(context)?;
+    let context_str = args_iter
+        .next()
+        .ok_or_else(|| ErrorItem::Internal(InternalError::new()))?
+        .get_name_or_string(context)?;
+    let sid_context = match Context::try_from(context_str.to_string()) {
+        Ok(c) => c,
+        Err(_) => {
+            return Err(CascadeErrors::from(
+                ErrorItem::make_compile_or_internal_error(
+                    "Invalid context",
+                    Some(file),
+                    context_str.get_range(),
+                    "Cannot parse this into a context",
```

Comment:
I looked around a bit on this one and we do this exact thing with Context::try_from in a few places.  But I feel like we should be passing up some info in the error we catch here that will help the user out a bit more.  Things like invalid user, role, type, mls at the very least.

I don't vote we fix it in this PR but should go on the TODO pile.

Reply:
#171